net_util/src/tap.rs

```diff
@@ -176,6 +179,10 @@ impl Tap {
 
     /// Set mac addr for tap interface.
     pub fn set_mac_addr(&self, addr: MacAddr) -> Result<()> {
+        if self.get_mac_addr()? == addr {
```

Comment:
I guess this means the user will have to provide the MAC address corresponding to the TAP interface, otherwise this would fallback onto setting the randomly generated one?
I'm asking because this doesn't feel natural for the user. By default, if the user provides the TAP interface, he should simply need to provide the name.

Reply:
The host mac is always saved into the configuration even if it was based on the randomly assigned one from the Linux kernel. Upon reboot it will look like there is a host mac assigned so it will try and apply it, this check prevents that as the tap interface already has that address (it's where it came from originally.)

We need to always save the host mac in order to support snapshot/restore.

Reply:
I've added a bit of a comment!